### PR TITLE
Implement Get/Set TableServiceProperties

### LINKED ISSUE #11925: Tables: Implement Get Table Service Properties support
# Summary
The Get Table Service Properties operation gets the properties of a storage account’s Table service, including properties for Storage Analytics and CORS (Cross-Origin Resource Sharing) rules. Note, these capabilities are only supported by storage endpoints.

# Scope of Work
- Implement support in the rest client to support get table service properties. 
- Implement a public client method to wire up get service properties to the rest client.
- Implement and record live tests covering this API method

# Out of Scope
Any modification to the swagger or generator to auto-generate support in the rest client

# Success Criteria
- The `StorageServiceProperties` payload can be retrieved from the service

# Related Issues and References
https://docs.microsoft.com/en-us/rest/api/storageservices/get-table-service-properties
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/tables/Azure.Data.Tables/src/TableInternalRestClient.cs
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/tables/Azure.Data.Tables/src/TableInternalClient.cs
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/tables/Azure.Data.Tables/src/TableClient.cs

### PR DESCRIPTION
**Pull request by meanjeekim**
Fixes #11925 + #11924 

- Implemented get/set TableServiceProperties
- Wrote and recorded test
- Updated api

### CODE REVIEW

File: sdk/tables/Azure.Data.Tables/tests/TableServiceClientLiveTests.cs [hunk 1]
```diff
@@ -21,7 +21,7 @@ namespace Azure.Data.Tables.Tests
     public class TableServiceClientLiveTests : TableServiceLiveTestsBase
     {
 
-        public TableServiceClientLiveTests(bool isAsync) : base(isAsync /* To record tests, add this argument, RecordedTestMode.Record */)
+        public TableServiceClientLiveTests(bool isAsync) : base(isAsync, RecordedTestMode.Record /* To record tests, add this argument, RecordedTestMode.Record */)
```
Comment (tg-msft) [inline sdk/tables/Azure.Data.Tables/tests/TableServiceClientLiveTests.cs:24]: Looks like leaving this in accidentally is causing the test run to fail.

File: sdk/tables/Azure.Data.Tables/tests/TableServiceClientLiveTests.cs [hunk 2]
```diff
@@ -95,5 +95,68 @@ public async Task GetTablesReturnsTablesWithFilter()
...
+            IList<CorsRule> corsRules = new List<CorsRule>();
+
+            // Create TableServiceProperties to set
+            var tableServicePropertiesToSet = new TableServiceProperties(
+                new LoggingSettings("1.0", !beforeResponse.Logging.Delete, true, true, new RetentionPolicy(false)), hourMetrics: new TableMetrics("1.0", true, true, new RetentionPolicy(true, 7)), minuteMetrics: new TableMetrics("1.0", false, null, new RetentionPolicy(false)), corsRules);
+
+            // Set properties
+            await service.SetPropertiesAsync(tableServicePropertiesToSet).ConfigureAwait(false);
+
+            // Wait 20 seconds if on Live mode
```
Comment (maririos) [inline sdk/tables/Azure.Data.Tables/tests/TableServiceClientLiveTests.cs:115]: could we add a comment here on the why? Is it a recommended waiting time?
I wonder if there is another way to see if the properties are propagated and ready to consume.

**Review by christothes — COMMENTED** [review]
Looks good, I just added a few comments and suggestions,

File: sdk/tables/Azure.Data.Tables/tests/TableServiceClientLiveTests.cs (hunk 2 above)
Comment (meanjeekim) [reply sdk/tables/Azure.Data.Tables/tests/TableServiceClientLiveTests.cs:115]: It was around the minimum amount of time for the property changes to be made for the sync methods.

**Review by christothes — APPROVED** [review]
👍

**Pull request merged**